pgmax/fg/graph.py

```diff
@@ -304,6 +369,38 @@ def __post_init__(self):
                 object.__setattr__(self, field, MappingProxyType(getattr(self, field)))
 
 
+@dataclass(frozen=True, eq=False)
+class GraphWiring:
```

Comment:
A more generic structure (e.g. a dictionary) might be better than a new dedicated class (which we have to update every time we add a new factor type).

Reply:
Agreed

Reply:
Get rid of this class. Not necessary